lib/babe/build_integration_test.go

```diff
@@ -242,77 +241,66 @@ func TestBuildAndApplyExtrinsic(t *testing.T) {
-	gen, genTrie, genHeader := newWestendLocalGenesisWithTrieAndHeader(t)
-	babeService := createTestService(t, ServiceConfig{}, gen, genTrie, genHeader)
-
-	// see https://github.com/noot/substrate/blob/add-blob/core/test-runtime/src/system.rs#L468
-	// add a valid transaction
-	txa := []byte{
-		3, 16, 110, 111, 111, 116,
-		1, 64, 103, 111, 115, 115,
-		97, 109, 101, 114, 95, 105,
-		115, 95, 99, 111, 111, 108}
-	vtx := transaction.NewValidTransaction(types.Extrinsic(txa), &transaction.Validity{})
-	babeService.transactionState.Push(vtx)
+func TestBuildAndApplyExtrinsic_InvalidPayment(t *testing.T) {
+	genesis, genesisTrie, genesisHeader := newWestendDevGenesisWithTrieAndHeader(t)
+	babeService := createTestService(t, ServiceConfig{}, genesis, genesisTrie, genesisHeader)
 
-	// add a transaction that can't be included (transfer from account with no balance)
-	// See https://github.com/paritytech/substrate/blob/5420de3face1349a97eb954ae71c5b0b940c31de/core/transaction-pool/src/tests.rs#L95
-	txb := []byte{
-		1, 212, 53, 147, 199, 21, 253, 211,
-		28, 97, 20, 26, 189, 4, 169, 159,
-		214, 130, 44, 133, 88, 133, 76, 205,
-		227, 154, 86, 132, 231, 165, 109, 162,
-		125, 142, 175, 4, 21, 22, 135, 115, 99,
-		38, 201, 254, 161, 126, 37, 252, 82,
-		135, 97, 54, 147, 201, 18, 144, 156,
-		178, 38, 170, 71, 148, 242, 106, 72,
-		69, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-		0, 0, 0, 0, 216, 5, 113, 87, 87, 40,
-		221, 120, 247, 252, 137, 201, 74, 231,
-		222, 101, 85, 108, 102, 39, 31, 190, 210,
-		14, 215, 124, 19, 160, 180, 203, 54,
-		110, 167, 163, 149, 45, 12, 108, 80,
-		221, 65, 238, 57, 237, 199, 16, 10,
-		33, 185, 8, 244, 184, 243, 139, 5,
-		87, 252, 245, 24, 225, 37, 154, 163, 142}
-	vtx = transaction.NewValidTransaction(types.Extrinsic(txb), &transaction.Validity{})
-	babeService.transactionState.Push(vtx)
+	parentHash := common.MustHexToHash("0x35a28a7dbaf0ba07d1485b0f3da7757e3880509edc8c31d0850cb6dd6219361d")
+	header := types.NewHeader(parentHash, common.Hash{}, common.Hash{}, 1, types.NewDigest())
 
-	zeroHash, err := common.HexToHash("0x00")
+	bestBlockHash := babeService.blockState.BestBlockHash()
+	rt, err := babeService.blockState.GetRuntime(bestBlockHash)
 	require.NoError(t, err)
 
-	parentHeader := &types.Header{
-		ParentHash: zeroHash,
-	}
+	err = rt.InitializeBlock(header)
+	require.NoError(t, err)
 
-	duration, err := time.ParseDuration("1s")
+	rawMeta, err := rt.Metadata()
+	require.NoError(t, err)
+	var decoded []byte
+	err = scale.Unmarshal(rawMeta, &decoded)
 	require.NoError(t, err)
 
-	slot := Slot{
-		start:    time.Now(),
-		duration: duration,
-		number:   1000,
-	}
+	meta := &ctypes.Metadata{}
+	err = codec.Decode(decoded, meta)
+	require.NoError(t, err)
 
-	bestBlockHash := babeService.blockState.BestBlockHash()
-	rt, err := babeService.blockState.GetRuntime(bestBlockHash)
+	rv := rt.Version()
+	bob, err := ctypes.NewMultiAddressFromHexAccountID(
+		"0x90b5ab205c6974c9ea841be688864633dc9ca8a357843eeacf2314649965fe22")
```

Comment:
isn't this stored somewhere in our codebase?

Reply:
yup, updated